client/blocks/nps-survey/index.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import RecommendationSelect from './recommendation-select';
+
+class NpsSurvey extends Component {
+	static propTypes = {
+		onDismissed: PropTypes.func,
+		name: PropTypes.string
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			recommendationValue: null,
+			isSubmitting: false,
+			isSubmitted: false
+		};
+		this.handleRecommendationSelectChange = this.handleRecommendationSelectChange.bind( this );
+		this.handleFinishClick = this.handleFinishClick.bind( this );
+		this.handleDismissClick = this.handleDismissClick.bind( this );
+	}
+
+	handleRecommendationSelectChange( newRecommendationValue ) {
+		this.setState( {
+			recommendationValue: newRecommendationValue
+		} );
+	}
+
+	handleFinishClick() {
+		// TODO: fire Redux action and use Redux state tree to determine when
+		// survey has been submitted
+		this.setState( {
+			isSubmitting: true
+		} );
+
+		// simulate requestresponse time
+		setTimeout( () => {
+			this.setState( {
+				isSubmitted: true
```

Comment:
I'd also set `isSubmitting` to false here & disable the buttons below when:
`this.state.isSubmitting || this.state.isSubmitted`

...but this is functionally equivalent